src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,164 @@
+                        if (!model.hasPerson(student)) {
+                            new AddStudentCommand(student).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Student ID for student category");
+                    }
+                    break;
+
+                case "company":
+                    String industry = values[2].trim();
+                    if (!industry.isEmpty()) {
+                        Company company = new Company(name, new Industry(industry), phone, email, address, tagSet);
+                        if (!model.hasPerson(company)) {
+                            new AddCompanyCommand(company).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Industry for company category");
+                    }
+                    break;
+
+                default:
+                    throw new CommandException(MESSAGE_INVALID_CATEGORY);
+                }
+            }
+
+            return new CommandResult(String.format(MESSAGE_SUCCESS, filePath)
+                    + "\nSuccessfully imported: " + successCount + " entries");
+
+        } catch (IOException e) {
+            throw new CommandException(String.format(MESSAGE_FAILURE, filePath));
+        }
+    }
+
+    /**
+     * Parses a comma-separated string of tags into a {@code Set<Tag>}.
+     *
+     * @param tagsString The comma-separated string of tags.
+     * @return A {@code Set<Tag>} containing the parsed tags.
+     */
+    private Set<Tag> parseTags(String tagsString) {
+        Set<Tag> tags = new HashSet<>();
+        if (!tagsString.isEmpty()) {
+            String[] tagsArray = tagsString.split("\\s+"); // Split tags by spaces
+            for (String tagName : tagsArray) {
+                tags.add(new Tag(tagName));
+            }
+        }
+        return tags;
+    }
+    /**
+     * Checks if two ImportCommand objects are equal.
+     * Two ImportCommand objects are considered equal if they have the same file path.
+     *
+     * @param other The other object to compare with.
+     * @return true if both objects have the same file path, false otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof ImportCommand)) {
+            return false;
+        }
+
+        ImportCommand otherCommand = (ImportCommand) other;
+        return filePath.equals(otherCommand.filePath);
+    }
+
+    /**
+     * Returns the hash code for this ImportCommand.
+     *
+     * @return The hash code for this ImportCommand, based on the file path.
+     */
+    @Override
+    public int hashCode() {
+        return filePath.hashCode();
```

Comment:
Can maybe add a test case to check if the actual `hashCode` returned is equal with the expected one